### PR TITLE
fix the bug of adamw which set the attribute in param group not working

### PR DESCRIPTION
**Pull request by GuoxiaWang**
### PR types

Bug fixes
### PR changes

APIs
### Describe

exclude `learning_rate`,  set the attributes in param group will overwrite the attributes set in outer.
The `learning_rate = 0.1 * 0.5 = 0.05` used to update the param for `embedding_m.weight`

```
    optimizer = paddle.optimizer.AdamW(
        learning_rate=0.1,
        parameters=[{
            'params': embedding_l.weight,
        }, {
            'params': embedding_m.weight,
            'weight_decay': 0.0001,
            'learning_rate': 0.5,
            'beta1': 0.8
        }],
        beta1=0.9,
        beta2=0.98,
        epsilon=1e-6,
        weight_decay=0.1)
```

### CODE REVIEW

File: python/paddle/optimizer/adamw.py [hunk 1]
```diff
@@ -25,7 +31,7 @@
 __all__ = []
 
 
-class AdamW(Adam):
+class AdamW(Optimizer):
     r"""
     The AdamW optimizer is implemented based on the AdamW Optimization
```
Comment (minogame) [inline python/paddle/optimizer/adamw.py:36]: Now it is based on the base Optimizer class. Should have cleaned the documents.

File: python/paddle/optimizer/adamw.py [hunk 2]
```diff
@@ -160,37 +171,108 @@ def __init__(self,
...
             raise ValueError("Invaild value of epsilon, expect epsilon >= 0.")
-        coeff = weight_decay
-        if not isinstance(coeff, float) and \
-                not isinstance(coeff, framework.Variable):
-            raise TypeError("coeff should be float or Tensor.")
-        self._params_name = set()
-        self._apply_decay_param_fun = apply_decay_param_fun
-        self._coeff = coeff
-        self._lr_to_coeff = dict()
+        if not isinstance(weight_decay, float) and \
```
Comment (minogame) [inline python/paddle/optimizer/adamw.py:174]: Suggest to check if the type is int, for some cases the user may use `weight_decay = 0`

File: python/paddle/optimizer/adamw.py [hunk 3]
```diff
@@ -160,37 +171,108 @@ def __init__(self,
...
+            # list(paddle.Tensor) will be a error value
+            if isinstance(parameters, (paddle.Tensor, core.eager.Tensor)):
+                raise TypeError(
+                    "`parameters` argument given to the optimizer should be "
+                    "an iterable of paddle Tensors, but got argument type is `{}`.".
+                    format(type(parameters)))
+            if isinstance(parameters, dict):
+                raise TypeError(
+                    "`parameters` argument should not get dict type, "
+                    "if parameter groups is needed, please set `parameters`"
```
Comment (minogame) [inline python/paddle/optimizer/adamw.py:195]: if parameter groups **are** needed

**Review by minogame — COMMENTED** [review]
provided some trivial suggestions.

The fix now works in my codes.

File: python/paddle/optimizer/adamw.py (hunk 3 above)
Comment (GuoxiaWang) [reply python/paddle/optimizer/adamw.py:195]: It means parameters need a list of dict.

**Review by TCChenlong — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**